composition-js/src/merging/merge.ts

```diff
@@ -1709,6 +1716,37 @@ class Merger {
         continue;
       }
 
+      const fromContextDirective = this.subgraphs.values()[idx].metadata().fromContextDirective();
+
+      const contextArguments = (source.kind === 'FieldDefinition' ? source.arguments() : [])
+        .map((arg): {
+          context: string,
+          name: string,
+          type: string,
+          selection: string,
+        } | undefined => {
+          if (!isFederationDirectiveDefinedInSchema(fromContextDirective)) {
+            return undefined;
+          }
+          const appliedDirectives = arg.appliedDirectivesOf(fromContextDirective);
+          if (appliedDirectives.length === 0) {
+            return undefined;
+          }
+          assert(appliedDirectives.length === 1, 'There should be at most one @fromContext directive applied to an argument');
+          const directive = appliedDirectives[0];
+          const { context, selection } = parseContext(directive.arguments().field);
+          // these are validated in the subgraph validation phase
+          assert(context, 'Context should be defined');
+          assert(selection, 'Selection should be defined');
+          return {
+            context: `${this.subgraphs.values()[idx].name}__${context}`,
```

Comment:
We do need to use this in the future to get the variable name, so at some point it has to be valid GraphQL. At one point I was randomly generating those variable names, but I decided for observability reasons it would be better to keep them consistent. I'll opt for forbidding a leading '__'